e2e/security/security.go

```diff
@@ -0,0 +1,213 @@
+			opts:       []string{"--security", "seccomp:./security/testdata/seccomp-profile.json"},
+			expectExit: 159, // process should be killed with SIGSYS (128+31)
+		},
+		{
+			name:       "SecComp_true",
+			image:      c.env.ImagePath,
+			argv:       []string{"true"},
+			opts:       []string{"--security", "seccomp:./security/testdata/seccomp-profile.json"},
+			expectExit: 0,
+		},
+		// capabilities
+		{
+			name:       "capabilities_keep_true",
+			image:      c.env.ImagePath,
+			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:       []string{"--keep-privs"},
+			expectExit: 255,
+		},
+		{
+			name:       "capabilities_keep-false",
+			image:      c.env.ImagePath,
+			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
+			expectExit: 2,
+		},
+		{
+			name:       "capabilities_drop",
+			image:      c.env.ImagePath,
+			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:       []string{"--drop-caps", "CAP_NET_RAW"},
+			expectExit: 2,
+		},
+	}
+
+	for _, tt := range tests {
+		optArgs := []string{}
+		optArgs = append(optArgs, tt.opts...)
+		optArgs = append(optArgs, c.env.ImagePath)
+		optArgs = append(optArgs, tt.argv...)
+
+		c.env.RunSingularity(
+			t,
+			e2e.AsSubtest(tt.name),
+			e2e.WithCommand("exec"),
+			e2e.WithArgs(optArgs...),
+			e2e.ExpectExit(tt.expectExit),
+		)
+
+	}
+}
+
+// testSecurityPriv tests security flag fuctionality for singularity exec with elevated privileges
+func (c *ctx) testSecurityPriv(t *testing.T) {
+	tests := []struct {
+		name       string
+		argv       []string
+		opts       []string
+		expectOut  string
+		expectExit int
+	}{
+		// taget UID/GID
+		{
+			name:       "Set_uid",
+			argv:       []string{"id", "-u"},
+			opts:       []string{"--security", "uid:99"},
+			expectOut:  "99",
+			expectExit: 0,
+		},
+		{
+			name:       "Set_gid",
+			argv:       []string{"id", "-g"},
+			opts:       []string{"--security", "gid:99"},
+			expectOut:  "99",
+			expectExit: 0,
+		},
+		// seccomp from json file
+		{
+			name:       "SecComp_BlackList",
+			argv:       []string{"mkdir", "/tmp/foo"},
+			opts:       []string{"--security", "seccomp:./testdata/seccomp-profile.json"},
+			expectExit: 159, // process should be killed with SIGSYS (128+31)
+		},
+		{
+			name:       "SecComp_true",
+			argv:       []string{"true"},
+			opts:       []string{"--security", "seccomp:./testdata/seccomp-profile.json"},
+			expectExit: 0,
+		},
+		// capabilities
+		{
+			name:       "capabilities_keep",
+			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:       []string{"--keep-privs"},
+			expectExit: 0,
+		},
+		{
+			name:       "capabilities_drop",
+			argv:       []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:       []string{"--drop-caps", "CAP_NET_RAW"},
+			expectExit: 2,
+		},
+	}
+
+	for _, tt := range tests {
+		optArgs := []string{}
+		optArgs = append(optArgs, tt.opts...)
+		optArgs = append(optArgs, c.env.ImagePath)
+		optArgs = append(optArgs, tt.argv...)
+
```

Comment:
I will apply a quick fix to ignore nil pointer in `resultOps` of `ExpectExit`, so you could get rid of change in `singularitycmd.go` and instead use something more flexible with :


```
tests := []struct {
	name       string
	argv       []string
	opts       []string
	expectOp   e2e.SingularityCmdResultOp
	expectExit int
}{
	// taget UID/GID
	{
		name:       "Set_uid",
		argv:       []string{"id", "-u"},
		opts:       []string{"--security", "uid:99"},
		expectOp:   e2e.ExpectOutput(e2e.ExactMatch, "99"),
		expectExit: 0,
	},
...
}

c.env.RunSingularity(
	t,
	e2e.AsSubtest(tt.name),
	e2e.WithPrivileges(true),
	e2e.WithCommand("exec"),
	e2e.WithArgs(optArgs...),
	e2e.ExpectExit(tt.expectExit, tt.expectOp),
)
```

Reply:
Thank You! :+1: